src/main/kotlin/traffic_simulation/RoadNetwork.kt

```diff
@@ -40,30 +36,25 @@ class RoadNetwork(val capacity : Int){
     }
 
 
-    fun scenario(vehicleList: List<Vehicle>): MutableList<Vehicle>{
-        val vehiclesDriving : MutableList<Vehicle> = mutableListOf()
-
-        val vehiclesPlanningToDrive: MutableList<Vehicle> = gatherPlansToDrive(vehicleList)
-
-        val demand : Int = calculateDemand(vehiclesPlanningToDrive)
+    fun simulateInternScenarioAndPrintToConsole(vehicleList: List<Vehicle>){
```

Comment:
Ich antworte mal etwas out of order:

> außer vielleicht du willst darauf hinaus, dass ich jedes Szenario mit der gleichen Funktion simulieren soll, unabhängig von der Listenherkunft.

Jep. Warum sollte es für die Simulation interessant sein, ob die Liste im Code zusammengestellt, aus einer CSV-Datei, einer Datenbank oder dem Internet kommt? :smiley: Die Simulation funktioniert immer identisch.

> Die print-Anweisung dient der Ergebnisausgabe ohne csv Datei. 

Es klingt einleuchtend, dass du bei Ausgabe in CSV vielleicht nicht printen willst. Dann würde ich den Teil hier rausnehmen und dort einfügen, wo du es brauchst.

> Demnach hätte ich mir eine der sim Funktionen sparen können(?).

Meinst du die `sim` Funktionen in `Main`? Da kann man sich keine ohne weiteres sparen, denn die Abläufe sind ja unterschiedlich. Aus CSV: CSV parsen, simulieren, CSV schreiben. Aus fixer Liste: Liste zusammenstellen, simulieren, Ergebnisse printen. (Beachte, dass der "simulieren" Teil immer identisch ist.)